packages/website/ts/components/wallet/wallet.tsx

```diff
@@ -434,41 +457,56 @@ export class Wallet extends React.Component<WalletProps, WalletState> {
     }
     private async _fetchBalancesAndAllowancesAsync(tokenAddresses: string[]) {
-        const trackedTokenStateByAddress = this.state.trackedTokenStateByAddress;
+        const balanceAndAllowanceTupleByAddress: ItemByAddress<BigNumber[]> = {};
         const userAddressIfExists = _.isEmpty(this.props.userAddress) ? undefined : this.props.userAddress;
         for (const tokenAddress of tokenAddresses) {
-            const [balance, allowance] = await this.props.blockchain.getTokenBalanceAndAllowanceAsync(
+            const balanceAndAllowanceTuple = await this.props.blockchain.getTokenBalanceAndAllowanceAsync(
                 userAddressIfExists,
                 tokenAddress,
             );
-            trackedTokenStateByAddress[tokenAddress] = {
-                balance,
-                allowance,
-                isLoaded: true,
-            };
+            balanceAndAllowanceTupleByAddress[tokenAddress] = balanceAndAllowanceTuple;
         }
+        const pricesByAddress = await this._getPricesByAddressAsync(tokenAddresses);
+        const trackedTokenStateByAddress = _.reduce(
+            tokenAddresses,
+            (acc, address) => {
+                const [balance, allowance] = balanceAndAllowanceTupleByAddress[address];
+                const price = pricesByAddress[address];
+                acc[address] = {
+                    balance,
+                    allowance,
+                    price,
+                    isLoaded: true,
+                };
+                return acc;
+            },
+            this.state.trackedTokenStateByAddress,
+        );
+
         if (!this._isUnmounted) {
             this.setState({
                 trackedTokenStateByAddress,
             });
         }
     }
     private async _refetchTokenStateAsync(tokenAddress: string) {
-        const userAddressIfExists = _.isEmpty(this.props.userAddress) ? undefined : this.props.userAddress;
-        const [balance, allowance] = await this.props.blockchain.getTokenBalanceAndAllowanceAsync(
-            userAddressIfExists,
-            tokenAddress,
-        );
-        this.setState({
-            trackedTokenStateByAddress: {
-                ...this.state.trackedTokenStateByAddress,
-                [tokenAddress]: {
-                    balance,
-                    allowance,
-                    isLoaded: true,
-                },
-            },
-        });
+        await this._fetchBalancesAndAllowancesAsync([tokenAddress]);
+    }
+    private async _getPricesByAddressAsync(tokenAddresses: string[]): Promise<ItemByAddress<BigNumber>> {
+        if (_.isEmpty(tokenAddresses)) {
+            return {};
+        }
+        const tokenAddressesQueryString = tokenAddresses.join(',');
+        const endpoint = `${configs.BACKEND_BASE_URL}/prices?tokens=${tokenAddressesQueryString}`;
+        const response = await fetch(endpoint);
+        if (response.status !== 200) {
+            return {};
```

Comment:
Let's perhaps log more information about what went wrong to the console? If it was an error, let's also log to Rollbar.

Reply:
Ok, I looked a bit more and we can improve / unify our error handling when requesting the `/wiki` an `/eth_gas_station` endpoints as well. Let's merge this as is and I'll follow up today with a unified object for hitting our backend